File: c_files/n-body.c
#include <stdio.h>

int main() {
    int total = 0;
    int n = 50000000;
    
    for(int i = 0; i < n; i++) {
        float a = i * 3.14159f;
        float b = a / 2.71828f;
        float c = a + b - i;
        
        int temp = (int)c;
        int modulo = temp - (temp / 1000) * 1000;
        total = total + modulo;
    }
    return total;
}

File: rust_files/n-body.rs
use std::hint::black_box;

fn main() {
    let mut total: i32 = 0;
    let n: i32 = 50000000;
    
    for i in 0..n {
        let a: f32 = i as f32 * 3.14159;
        let b: f32 = a / 2.71828;
        let c: f32 = a + b - i as f32;
        
        let temp: i32 = c as i32;
        let modulo: i32 = temp - (temp / 1000) * 1000;
        total += modulo;
    }
    
    black_box(total); 
}
